pkg/util/pki/csr.go

```diff
@@ -95,6 +99,22 @@ func IPAddressesToString(ipAddresses []net.IP) []string {
 	return ipNames
 }
 
+func IPAddressesFromStrings(ipStrings []string) ([]net.IP, error) {
```

Comment:
This new function will return a meaningful error when the IP is invalid and cannot be parsed.

Reply:
We already disallow invalid IPs in the Webhook, so this should not be breaking: https://github.com/cert-manager/cert-manager/blob/4209de23716562f44f3f7295b1f162bbb69f6ccd/internal/apis/certmanager/validation/certificate.go#L192-L204